oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGCRecommendations.java

```diff
@@ -101,6 +108,22 @@ public VersionGCRecommendations(long maxRevisionAgeMs, Checkpoints checkpoints,
         TimeInterval scope = new TimeInterval(oldestPossible, Long.MAX_VALUE);
         scope = scope.notLaterThan(keep.fromMs);
 
+        detailedGCTimestamp = settings.get(SETTINGS_COLLECTION_DETAILED_GC_TIMESTAMP_PROP);
+        if (detailedGCTimestamp == 0) {
+            if (log.isDebugEnabled()) {
+                log.debug("No detailedGCTimestamp found, querying for the oldest deletedOnce candidate");
+            }
+            oldestPossibleFullGC = vgc.getOldestModifiedTimestamp(clock) - 1;
+            if (log.isDebugEnabled()) {
+                log.debug("detailedGCTimestamp found: {}", Utils.timestampToString(oldestPossibleFullGC));
+            }
+        } else {
+            oldestPossibleFullGC = detailedGCTimestamp - 1;
+        }
+
+        TimeInterval fullGCTimeInternal = new TimeInterval(oldestPossibleFullGC, MAX_VALUE);
```

Comment:
`TimeInterval` here expects milliseconds - however in the current version of the code `oldestPossibleFullGC` is seconds - as it comes from `getOldestModifiedTimestamp` which is based on `_modified`.

I would suggest to use milliseconds consistently.

Reply:
`getOldestModifiedTimestamp` is overridden in `MongoVersionGCSupport` which indeed converts the `_modified` value to `millis`.

Reply:
Ah, I see your point now, the `getOldestModifiedTimestamp` inside `VersionGCSupport` return it from `_modified` and it's in seconds.